src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -1340,25 +1340,21 @@ public Tensor pin_memory()
             /// This attribute is null by default and becomes a Tensor the first time a call to backward() computes gradients for the tensor.
             /// The attribute will then contain the gradients computed and future calls to backward() will accumulate (add) gradients into it.
             /// </summary>
-            public Tensor? grad()
-            {
-                var res = NativeMethods.THSTensor_grad(Handle);
-                CheckForErrors();
-
-                if (res == IntPtr.Zero)
-                    return null;
+            public Tensor? grad {
+                get {
+                    var res = NativeMethods.THSTensor_grad(Handle);
+                    CheckForErrors();
```

Comment:
I know it was the same in the old code, but i think we should check for 'res == IntPtr.Zero' before checking for errors.

Reply:
It's [line 242](https://github.com/dotnet/TorchSharp/pull/1299/files/5ac8a73e5f5247791dcdc3e6cae1d9cbf9bcf9eb#diff-c9fe99719d1977951a66f323175f5bcd3ba449f30a3711bbc84defa26e924460R242) disposing that.

Reply:
I have modified it in [this commit](https://github.com/dotnet/TorchSharp/pull/1299/commits/4e88b62ab2f40e3648e3b6fc6d5e8fec88e136c4).